sematic/graph.py

```diff
@@ -384,7 +382,7 @@ def _get_skip_reset_run_ids(
     @memoized_indexed
     def _get_artifact_value(self, artifact_id: str) -> Any:
         artifact = self._artifacts_by_id[artifact_id]
-        return get_artifact_value(artifact, self.storage)
+        return api_client.get_artifact_value(artifact)
```

Comment:
1. is this true? `self._artifacts_by_id[artifact_id] ===  api_client.get_artifact_value(artifact)` If not, why?
2. If we have artifact_id, I guess we can directly call `api_client. get_artifact_value_by_id(artifact_id)`.

I guess at least one is true, so I am a bit confused.

Reply:
`self._artifacts_by_id` contains artifact objects (`Artifact` instances), not artifact values.
`self._artifacts_by_id` is a cache so that we don't re-fetch artifacts if they are accessed multiple times.

Using `get_artifact_value` saves a round-trip as `get_artifact_value_by_id` will need to fetch the artifact first. Since we already have it in the cache, this is more economical.

Reply:
I see. So artifact and artifact value are too different things. Thanks for the explanation!